addons/website_quote/views/website_quotation.xml

```diff
@@ -19,7 +19,7 @@
                     <tr>
                         <th>Products</th>
                         <th>Quantity</th>
-                        <th>Taxes</th>
+                        <th><span t-if="True in [len(line.tax_id) > 0 for line in quotation.order_line]">Taxes</span></th>
```

Comment:
use "any" here. like
t-if="any([line.tax_id for line in quotation.order_line])"